scripts/release/helpers/requirements.js

```diff
@@ -32,53 +33,72 @@ function checkBranchDiff () {
       `Please visit ${link} for instructions to install.`
     )
   }
+
+  const branchDiffConfigPath = join(getApplicationConfigPath('changelog-maker'), 'config.json')
+
+  if (!existsSync(branchDiffConfigPath)) {
+    const link = 'https://github.com/nodejs/changelog-maker?tab=readme-ov-file#development'
+    fatal(
+      'The "branch-diff" configuration file is missing.',
+      `Please visit ${link} for instructions to configure.`
+    )
+  }
+
+  const requiredScopes = ['public_repo']
+  const { token } = JSON.parse(readFileSync(branchDiffConfigPath, 'utf8'))
+
+  checkGitHubScopes(token, requiredScopes, 'branch-diff')
 }
 
 // Check that the `gh` CLI is installed and authenticated.
 function checkGitHub () {
-  if (!process.env.GITHUB_TOKEN && !process.env.GH_TOKEN) {
+  const token = process.env.GITHUB_TOKEN || process.env.GH_TOKEN
+  const requiredScopes = ['public_repo', 'read:org']
+
+  if (!token) {
     const link = 'https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-personal-access-token-classic'
 
     fatal(
-      'The GITHUB_TOKEN environment variable is missing.',
+      'The GITHUB_TOKEN | GH_TOKEN environment variable is missing.',
       `Please visit ${link} for instructions to generate a personal access token.`,
       `The following scopes are required when generating the token: ${requiredScopes.join(', ')}`
     )
   }
 
   try {
-    run('gh --version')
+    run('gh --version', true)
   } catch (e) {
     fatal(
       'The "gh" CLI could not be found.',
       'Please visit https://github.com/cli/cli#installation for instructions to install.'
     )
   }
 
-  checkGitHubScopes()
+  checkGitHubScopes(token, requiredScopes, 'GITHUB_TOKEN | GH_TOKEN')
 }
 
-// Check that the active GITHUB_TOKEN has the required scopes.
-function checkGitHubScopes () {
+function checkGitHubScopes (token, requiredScopes, source) {
   const url = 'https://api.github.com'
   const headers = [
     'Accept: application/vnd.github.v3+json',
-    `Authorization: Bearer ${process.env.GITHUB_TOKEN || process.env.GH_TOKEN}`,
+    `Authorization: Bearer ${token}`,
     'X-GitHub-Api-Version: 2022-11-28'
   ].map(h => `-H "${h}"`).join(' ')
 
-  const lines = capture(`curl -sS -I ${headers} ${url}`).trim().split(/\r?\n/g)
```

Comment:
why remove .trim() ? you sure it's useless ?

Reply:
the output of `capture` is already trimmed (you added that in #4897 😉)